### DISCUSSION TITLE
ECMA-419 for humans?

### DISCUSSION THREAD

**Discussion by tve** [Moddable-OpenSource/moddable#978] [General]
Is there a ECMA-419 introductory blog post or a tutorial somewhere? Or do I have to wade through the standards doc to try and make sense of it?

**Comment by phoddie** [comment]
[ECMA-419](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/io/io.md) was written by humans for humans! ;)

That said, the primary audience is implementors of the APIs, not users of the APIs. That's similar to [ECMA-262](http://262.ecma-international.org) (JavaScript) which is written for engine implementors, not JavaScript developers. Neither one pretends to be an introduction to their respective topics.

There is an extensive introduction to ECMA-419 focused on I/O. It is in the Moddable SDK IO documentation [here](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/io/io.md). The documentation follows the initial implementation of ECMA-419 which was done on an ESP8266. It is probably slightly out of date on some of the details, but is accurate as an introduction for developers.
